src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -143,52 +142,85 @@ private static bool ApplyChangesFromTC31(JObject sarifLog)
                         }
                     }
 
-                    RecursivePropertyRename(run, "richText", "markdown");                    
+                    if (run["files"] is JArray files)
+                    {
+                        foreach (JObject artifact in files)
+                        {
+                            if (artifact["fileLocation"] is JObject location)
+                            {
+                                artifact.Remove("fileLocation");
+                                artifact["location"] = location;
+                            }
+                        }
+
+                        run["artifacts"] = files;
+                        run.Remove("files");
+                    }
+
+                    var universallyRenamedMembers = new Dictionary<string, string>
+                    {
+                        ["richText"] = "markdown",
+                        ["fileChange"] = "artifactChange",
+                        ["fileLocation"] = "artifactLocation",
+                        ["fileIndex"] = "artifactIndex",
```

Comment:
>["fileIndex"] = "artifactIndex" [](start = 24, length = 31)

Is this right? In the `artifact` (formerly `file`) object, you renamed `fileIndex` to `index`, not `artifactIndex`.